pac4j-oauth/src/main/java/org/pac4j/oauth/client/WeiboClient.java

```diff
@@ -0,0 +1,78 @@
+package org.pac4j.oauth.client;
+
+import org.pac4j.core.util.CommonHelper;
+import org.pac4j.oauth.exception.OAuthCredentialsException;
+import org.pac4j.oauth.profile.weibo.WeiboProfile;
+import org.pac4j.oauth.profile.weibo.WeiboProfileDefinition;
+import org.pac4j.scribe.builder.api.WeiboApi20;
+
+/**
+ * <p>This class is the OAuth client to authenticate users in Weibo using OAuth protocol version 2.0.</p>
+ * <p>The <i>scope</i> is by default : {@link WeiboScope#email}, but it can also but set to : {@link WeiboScope#all}
+ * or {@link WeiboScope#email}.</p>
+ * <p>It returns a {@link WeiboProfile}.</p>
+ * <p>More information at http://open.weibo.com/wiki/Oauth2/access_token/</p>
+ *
+ * @author Zhang Zhenli
+ * @since 3.1.0
+ */
+public class WeiboClient extends OAuth20Client<WeiboProfile> {
+
+    /**
+     * WeiboScope.
+     * <p>
+     * <p>More info at: <a href=
+     * "http://open.weibo.com/wiki/Scope">Scope</a></p>
+     */
+    public enum WeiboScope {
+        all,                        //     Request all of the following scope permissions
```

Comment:
I would have expected to see these constants in uppercase.

Reply:
ok